src/praisonai-agents/praisonaiagents/llm/failover.py

```
@@ -309,37 +321,39 @@ def mark_failure(
+                logger.warning(
+                    f"Profile '{profile.name}' error: {error}, "
+                    f"cooldown for {self.config.cooldown_on_error}s"
+                )
+            
+            # Notify callbacks
+            next_profile = self.get_next_profile()
+            if next_profile and next_profile != profile:
+                for callback in self._on_failover_callbacks:
+                    try:
+                        callback(profile, next_profile)
+                    except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)